kornia/geometry/transform/pyramid.py

```diff
@@ -254,23 +227,41 @@ def forward(self, x: torch.Tensor) -> Tuple[  # type: ignore
         return pyr, sigmas, pixel_dists
 
 
-# functional api
 def pyrdown(
         input: torch.Tensor,
         border_type: str = 'reflect', align_corners: bool = False) -> torch.Tensor:
     r"""Blurs a tensor and downsamples it.
 
     See :class:`~kornia.transform.PyrDown` for details.
     """
-    return PyrDown(border_type, align_corners)(input)
+    if not len(input.shape) == 4:
+        raise ValueError(f"Invalid input shape, we expect BxCxHxW. Got: {input.shape}")
+    kernel: torch.Tensor = _get_pyramid_gaussian_kernel()
+    # blur image
+    x_blur: torch.Tensor = filter2D(input, kernel, border_type)
+
+    # downsample.
+    out: torch.Tensor = F.interpolate(x_blur, scale_factor=0.5, mode='bilinear',
+                                      align_corners=align_corners)
```

Comment:
I remember pytorch throwing a warning around scale factor and size parameters. Could you take a look at it and figure what would be the proper way to use it ?